spec/support/http_handling_shared.rb

```diff
@@ -148,6 +148,18 @@
         expect(sockets_used.uniq.length).to eq(1)
       end
 
+      context "on a mixed state" do
+        it "re-opens the connection" do
+          first_socket = client.get("#{server.endpoint}/socket/1").body.to_s
```

Comment:
With all respect but how `first_socket_id = <...>.body.to_s` makes it more clear? You are saving first socket's response body into a variable named `first_socket_id`. That makes as much sense as naming it `first_socket`. IMO even naming variables `a` and `b` will be less confusing in this case.